File: April 25, 2026.py
def decompress(s):

    word_list = []
    new_s = {}
    result = []

    for word in s.split(" "):
        if word.isdigit():
            index = int(word) - 1
            found = word_list[index]
            result.append(found)
            word_list.append(found)
        else:
            word_list.append(word)
            result.append(word)

    new_s = " ".join(result)

    return new_s

File: April 25, 2026.js
function decompress(str) {

  let wordList = [];
  let newStr = {};
  let result = [];

  for (let word of str.split(" ")) {
    
    if (/^\d+$/.test(word)) {
      let index = Number(word) - 1;
      let found = wordList[index];
      result.push(found);
      wordList.push(found);
    } else {
      wordList.push(word);
      result.push(word)
    }
  }

  newStr = result.join(" ");

  return newStr;
}